New Thermostat example (#3675)
icerpc/icerpc-csharp: examples/README.md

| [Secure](./Secure/)                   | Shows how to secure TCP connections with TLS.                                                                   |
| [Stream](./Stream/)                   | Shows how to stream data from a client to a server.                                                             |
| [Telemetry](./Telemetry/)             | Shows how to use the telemetry interceptor and middleware.                                                      |

> | [Thermostat](./Thermostat/)           | Shows how to send requests via an intermediary server; includes sending requests the "other way around", from a server to a client.|

Reviewer: I personally like the way this is written, but it would be nice to have it with fit better.
Both fit on my screen (it's too long) and fit within the markdown table (it pushes the column right) How about: 
```suggestion
| [Thermostat](./Thermostat/)           | Shows how to send requests via an intermediary server and how to send requests from a server to a client.       |
```

Author: You're supposed to read the rendered version, not the GitHub review diff. 

Then, your "formatting" update changes the semantics. The example includes this server->client requests. The big picture is requests through an intermediary server. Both are not at the same level.
